### PR TITLE
Improve tests

### PR DESCRIPTION
**Pull request by MikailBag**
IMHO updating tests currently is quite painful.

I propose using `insta` snapshot testing library.
Insta provides special tool for reviewing and "blessing" snapshots.

### CODE REVIEW

**Comment by MikailBag** [comment]
Tests failure seems strange:
```
[2020-07-14T18:35:24Z ERROR paperclip::v2::codegen::emitter] Missing name for definition: 'DefaultSchema { name: None, reference: Some("#/definitions/Pet"), title: None, description: None, data_type: None, format: None, properties: {}, items: None, enum_: [], extra_props: None, required: {}, cyclic: false }'

thread 'tests_pet::test_lib_root' panicked at 'codegen: Missing name for definition
Stack backtrace:
   0: anyhow::error::<impl core::convert::From<E> for anyhow::Error>::from
             at /home/mb/.cargo/registry/src/github.com-1ecc6299db9ec823/anyhow-1.0.31/src/backtrace.rs:26
   1: <T as core::convert::Into<U>>::into
             at /home/mb/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libcore/convert/mod.rs:560
   2: paperclip::v2::codegen::emitter::Emitter::def_ns_name::{{closure}}
             at ./src/v2/codegen/emitter.rs:125
   3: core::option::Option<T>::ok_or_else
             at /home/mb/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libcore/option.rs:563
   4: paperclip::v2::codegen::emitter::Emitter::def_ns_name
             at ./src/v2/codegen/emitter.rs:121
   5: paperclip::v2::codegen::emitter::Emitter::def_mod_path
             at ./src/v2/codegen/emitter.rs:189
   6: paperclip::v2::codegen::emitter::RequirementCollector<E>::collect_parameters
             at ./src/v2/codegen/emitter.rs:906
   7: paperclip::v2::codegen::emitter::RequirementCollector<E>::collect_from_operation
             at ./src/v2/codegen/emitter.rs:797
   8: paperclip::v2::codegen::emitter::RequirementCollector<E>::collect
             at ./src/v2/codegen/emitter.rs:740
   9: paperclip::v2::codegen::emitter::Emitter::generate
             at ./src/v2/codegen/emitter.rs:302
  10: test_codegen::CODEGEN_PET_LIB::{{closure}}
             at tests/test_codegen.rs:32
...
```

**Comment by wafflespeanut** [comment]
> IMHO updating tests currently is quite painful.

I agree. I'd planned to extract the strings into separate files (which should get rid of the recompilation), but then I never did. :sweat: 

> I propose using `insta` snapshot testing library.

I'm certainly open to trying this out.

File: core/src/v2/models.rs [hunk 1]
```diff
@@ -825,6 +824,13 @@ where
             Resolvable::Resolved { ref old, .. } => old.read().description().map(String::from),
         }
     }
+
+    pub fn read(&self) -> parking_lot::RwLockReadGuard<S> {
+        match self {
+            Resolvable::Raw(s) => s.read(),
+            Resolvable::Resolved { old, .. } => old.read(),
```
Comment (wafflespeanut) [inline core/src/v2/models.rs:831]: I think the failure is coming from this line. We should read the resolved schema, not the unresolved one.

Comment (wafflespeanut) [reply core/src/v2/models.rs:831]: I also wonder what's the motivation behind this addition. This type already implements `Deref` to `Arc<RwLock<S>>`, so we won't really need this additional method.

Comment (MikailBag) [reply core/src/v2/models.rs:831]: Oh, yes, thanks.
I added this function by accident, will remove.

**Comment by MikailBag** [comment]
Well, as a downside snapshots are pretty large (approximately 60K lines).
Is it acceptable?

**Comment by MikailBag** [comment]
Also it seems that cache grew too fat again.
One of builds timed out while restoring the cache.

**Comment by wafflespeanut** [comment]
> Is it acceptable?

Yeah, it's okay. Some of the generated files are huge. :sweat_smile: 

> One of builds timed out while restoring the cache.

This is happening a lot lately. I had to get rid of the cache every time. Let's just disable cache for now.

**Comment by wafflespeanut** [comment]
Let me know when this is ready for review.

**Comment by MikailBag** [comment]
I think it's ready for review.
First commit changes tests (and also replaces lazy-static -> once-cell)
Second actually adds snapshots.
Third disables caching.

**Pull request merged**